test/images/crd-conversion-webhook/converter/example_converter.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func isEmptyObject(Object *unstructured.Unstructured) bool {
+	if len(Object.Object) != 2 {
+		return false
+	}
+	if _, ok := Object.Object["kind"]; !ok {
+		return false
+	}
+	if _, ok := Object.Object["apiVersion"]; !ok {
+		return false
+	}
+	return true
+}
+func convertExampleCRD(Object *unstructured.Unstructured, toVersion string) (*unstructured.Unstructured, metav1.Status) {
+	glog.V(2).Info("converting crd")
+
+	convertedObject := Object.DeepCopy()
+	fromVersion := Object.GetAPIVersion()
+
+	// API Server may send a conversion request for an empty object to test the conversion stack before starting
```

Comment:
@liggitt This is a stack trace in case you are interested to see why:

```
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion.(*webhookConverter).ConvertToVersion at webhook_converter.go:177
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion.(*safeConverterWrapper).ConvertToVersion at converter.go:151
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.schemaCoercingConverter.ConvertToVersion at customresource_handler.go:794
<autogenerated>:2
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*codec).Encode at versioning.go:219
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.Encode at codec.go:46
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.CheckCodec at codec_check.go:31
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/cacher.NewCacherFromConfig at cacher.go:202
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry.StorageWithCacher.func1 at storage_factory.go:65
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/registry/generic/registry.(*Store).CompleteWithOptions at store.go:1385
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/registry/customresource.newREST at etcd.go:102
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/registry/customresource.NewStorage at etcd.go:44
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.(*crdHandler).getOrCreateServingInfoFor at customresource_handler.go:491
k8s.io/kubernetes/vendor/k8s.io/apiextensions-apiserver/pkg/apiserver.(*crdHandler).ServeHTTP at customresource_handler.go:217
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP at pathrecorder.go:248
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP at pathrecorder.go:234
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP at handler.go:154
<autogenerated>:2
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1 at authorization.go:64
net/http.HandlerFunc.ServeHTTP at server.go:1947
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1 at maxinflight.go:160
net/http.HandlerFunc.ServeHTTP at server.go:1947
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1 at impersonation.go:50
net/http.HandlerFunc.ServeHTTP at server.go:1947
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1 at authentication.go:81
net/http.HandlerFunc.ServeHTTP at server.go:1947
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1 at timeout.go:100
runtime.goexit at asm_amd64.s:2361
```

it is for a test that is not exists in this PR but the line numbers should match enough.

Reply:
It's the codec smoke test. Should we have a way to disable that or alternatively to provide a smoke test object instead of an empty one?

Reply:
Depends on how the other discussion goes, this may not be important for CRD webhook conversion. I and @liggitt think a conversion from an empty object in one version should result in an empty object in another version. Let's conclude that discussion before going into this path.